src/modules/schnorr_adaptor/main_impl.h

```diff
@@ -0,0 +1,286 @@
+    secp256k1_gej r0j;
+    secp256k1_ge pk;
+    secp256k1_ge r;
+    secp256k1_ge r0;
+    secp256k1_ge t;
+    unsigned char nonce32[32] = {0};
+    unsigned char pk_buf[32];
+    unsigned char seckey[32];
+    size_t size = 33;
+    size_t msglen = 32;
+    int ret = 1;
+
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secp256k1_ecmult_gen_context_is_built(&ctx->ecmult_gen_ctx));
+    ARG_CHECK(sig65 != NULL);
+    ARG_CHECK(msg32 != NULL);
+    ARG_CHECK(keypair != NULL);
+    ARG_CHECK(t33 != NULL);
+
+    if (noncefp == NULL) {
+        noncefp = secp256k1_adaptor_nonce_function_bip340;
+    }
+
+    ret &= secp256k1_keypair_load(ctx, &sk, &pk, keypair);
+
+    if (secp256k1_fe_is_odd(&pk.y)) {
+        secp256k1_scalar_negate(&sk, &sk);
+    }
+
+    /* d */
+    secp256k1_scalar_get_b32(seckey, &sk);
+    /* bytes_from_point(P) */
+    secp256k1_fe_get_b32(pk_buf, &pk.x);
+
+    ret &= !!noncefp(nonce32, msg32, seckey, &t33[1], pk_buf, bip340_algo, sizeof(bip340_algo), ndata);
+    /* k0 */
+    secp256k1_scalar_set_b32(&k, nonce32, NULL);
+    ret &= !secp256k1_scalar_is_zero(&k);
+    secp256k1_scalar_cmov(&k, &secp256k1_scalar_one, !ret);
+
+    /* R = k0*G */
+    secp256k1_ecmult_gen(&ctx->ecmult_gen_ctx, &rj, &k);
+    secp256k1_ge_set_gej(&r, &rj);
+
+    /* T = cpoint(T) */
+    ret &= !!secp256k1_eckey_pubkey_parse(&t, t33, 33);
+
+    /* R' = k*G + T, can use gej_add_ge_var since r and t aren't secret */
+    secp256k1_gej_add_ge_var(&r0j, &rj, &t, NULL);
+    secp256k1_ge_set_gej(&r0, &r0j);
+
+    /* We declassify R' to allow using it as a branch point. This is fine
+     * because R' is not a secret.  */
+    secp256k1_declassify(ctx, &r0, sizeof(r0));
+    secp256k1_fe_normalize_var(&r0.y);
+    if (secp256k1_fe_is_odd(&r0.y)) {
+        secp256k1_scalar_negate(&k, &k);
+    }
+
+    ret &= !!secp256k1_eckey_pubkey_serialize(&r0, sig65, &size, 1);
+
+    secp256k1_schnorrsig_challenge(&e, &sig65[1], msg32, msglen, pk_buf);
+    secp256k1_scalar_mul(&e, &e, &sk);
+    /* k + e * d */
+    secp256k1_scalar_add(&e, &e, &k);
+    secp256k1_scalar_get_b32(&sig65[33], &e);
+
+    secp256k1_memczero(sig65, 65, !ret);
+    secp256k1_scalar_clear(&k);
+    secp256k1_scalar_clear(&sk);
+    memset(seckey, 0, sizeof(seckey));
+
+    return ret;
+}
+
+int secp256k1_schnorr_adaptor_presign(const secp256k1_context *ctx, unsigned char *sig65, const unsigned char *msg32, const secp256k1_keypair *keypair, const unsigned char *t33, const unsigned char *aux_rand32) {
+    /* We cast away const from the passed aux_rand32 argument since we know the default nonce function does not modify it. */
+    return secp256k1_schnorr_adaptor_presign_internal(ctx, sig65, msg32, keypair, secp256k1_adaptor_nonce_function_bip340, t33, (unsigned char*)aux_rand32);
+}
+
+int secp256k1_schnorr_adaptor_extract_t(const secp256k1_context *ctx, unsigned char *t33, const unsigned char *sig65, const unsigned char *msg32, const secp256k1_xonly_pubkey *pubkey) {
```

Comment:
We don't need this func to be const-time (can be variable time). So, no need to use a `ret` variable and store intermediate failures (like the `adaptor_presig`, and `schnorrsig_sign` do). 

We can simply `return 0` when scalar overflow, pubkey parse failure, etc. You can refer to the `schnorrsig_verify`, and `ecdsa_adaptor_verify` APIs. They are implemented as variable time too.